web_monitoring/cli.py

```diff
@@ -60,6 +143,99 @@ def _filter_unchanged_versions(versions):
             yield version
 
 
+def _list_ia_versions_for_urls(url_patterns, from_date, to_date,
+                               skip_repeats=True, version_filter=None,
+                               client=None):
+    version_filter = version_filter or _is_page
+    client = client or ia.WaybackClient()
+    skipped = 0
+
+    for url in url_patterns:
+        ia_versions = client.list_versions(url,
+                                           from_date=from_date,
+                                           to_date=to_date,
+                                           skip_repeats=skip_repeats)
+        try:
+            for version in ia_versions:
+                if version_filter(version):
+                    yield version
+                else:
+                    skipped += 1
+                    logger.debug('Skipping URL "%s"', version.url)
+        except ValueError as error:
+            logger.warn(error)
```

Comment:
Might as well use [`logger.exception`](https://docs.python.org/3/library/logging.html#logging.exception) for this.

```py
except ValueError:
    logger.exception('Error while applying version_filter to URL %r', url)
```

The Traceback will be automatically captured by `logger.exception` and shown below the log message.

Reply:
Oh, this is really here to handle the error if there were no versions to list: https://github.com/edgi-govdata-archiving/web-monitoring-processing/blob/4bb08f886a14e686f7eb0c68f2d853673aa6bd62/web_monitoring/internetarchive.py#L377-L379

…which, well, probably shouldn’t actually be a `ValueError` at all, since it’s not really a problem with your input. Actually, I’m not really sure why we have an exception for this at all. Should we get rid of that?

Reply:
(Anyway, implicit point there is that this isn’t really an *exception* or unexpected scenario, so I don’t think `logger.exception` makes sense for it. But the code and the error classes involved really don’t make that clear.)